test/components/modal.spec.tsx

```diff
@@ -0,0 +1,146 @@
+import * as React from 'react';
+import {ClientRenderer, expect, selectDom, simulate, sinon, waitFor} from 'test-drive-react';
+import {ModalDemo} from '../../demo/components/modal-demo';
+import {Modal} from '../../src';
+
+describe('<Modal />', () => {
+    const clientRenderer = new ClientRenderer();
+    const bodySelect = selectDom(document.body);
+
+    afterEach(() => clientRenderer.cleanup());
+
+    describe('A typical use case for the modal:', function() {
+        it('is hidden at first, a user clicks on a button, the modal appears,' +
+            'and then the user clicks on the background and the model closes', async function() {
+            clientRenderer.render(<ModalDemo />);
+
+            await waitFor(() => expect(bodySelect('MODAL')).to.be.absent());
+
+            simulate.click(bodySelect('MODAL_BUTTON'));
+
+            await waitFor(() => expect(bodySelect('MODAL')).to.be.present());
+
+            simulate.click(bodySelect('MODAL')!);
+
+            await waitFor(() => expect(bodySelect('MODAL')).to.be.absent());
+        });
+    });
+
+    it('renders to the screen', async function() {
+        clientRenderer.render(<Modal isOpen={true} />);
+
+        await waitFor(() => expect(bodySelect('MODAL')!).to.be.present());
+    });
+
+    it('renders any children passed as props', async function() {
+        clientRenderer.render(
+            <Modal isOpen={true}>
+                <p data-automation-id="CHILD_1">child 1</p>
+                <p data-automation-id="CHILD_2">child 2</p>
+            </Modal>
+        );
+
+        await waitFor(() => {
+            expect(bodySelect('CHILD_1')).to.be.present();
+            expect(bodySelect('CHILD_2')).to.be.present();
+        });
+    });
+
+    it('takes the full width and height of the viewport and is centered in the viewport', async function() {
+        clientRenderer.render(<Modal isOpen={true} />);
+
+        function checkIfAlignedToScreen(element: Element) {
+            const rects = element.getBoundingClientRect();
+            return rects.top === 0 && rects.left === 0;
+        }
+
+        await waitFor(() => {
+            expect(checkIfAlignedToScreen(bodySelect('MODAL')!), 'The modal wasn\'t centered').to.equal(true);
+            expect(bodySelect('MODAL')!.clientHeight).to.equal(window.innerHeight);
+            expect(bodySelect('MODAL')!.clientWidth).to.equal(window.innerWidth);
+        });
+    });
+
+    it('renders one child in the center of the viewport', async function() {
+        clientRenderer.render(
+            <Modal isOpen={true}>
+                <p data-automation-id="CHILD_1">child 1</p>
+            </Modal>
+        );
+
+        await waitFor(() => {
+            const child = bodySelect('CHILD_1');
+            const modal = bodySelect('MODAL');
+            expect([child, modal]).to.be.horizontallyAligned('center', 1);
```

Comment:
This test is not working. CHILD_1's width is 100% so it is always horizontally aligned. I suggest adding an inline style with something like:
<p .... style={{width: '50px'}}>